### ISSUE TITLE
lzma fails to install with micropip

### ISSUE THREAD

**Issue by dopplershift** [pyodide/pyodide#3003]
On 0.21.1 (from within jupyterlite), trying to do `await micropip.install('_lzma')` gives:

```pytb
File /lib/python3.10/site-packages/packaging/requirements.py:104, in Requirement.__init__(self, requirement_string)
    102     req = REQUIREMENT.parseString(requirement_string)
    103 except ParseException as e:
--> 104     raise InvalidRequirement(
    105         f'Parse error at "{ requirement_string[e.loc : e.loc + 8]!r}": {e.msg}'
    106     )
    108 self.name: str = req.name
    109 if req.url:

InvalidRequirement: Parse error at "'_lzma'": Expected W:(0-9A-Za-z)
```

Alternatively, trying `await micropip.install('lzma')` gives:

```pytb
File /lib/python3.10/asyncio/futures.py:201, in Future.result(self)
    199 self.__log_traceback = False
    200 if self._exception is not None:
--> 201     raise self._exception
    202 return self._result

TypeError: exceptions must derive from BaseException
```

Doing `import pyodide_js; await pyodide_js.loadPackage('_lzma')` works fine.

**Comment by rth** [comment]
Thanks for the report @dopplershift !

We do want these to be installable with micropip, right? cc @ryanking13  Or was the original intention to avoid shadowing identically named PyPI packages?

In any case, it looks like on dev we are back to lzma without the underscore and I would be +1 to also backport it to 0.21.2
Same for ssl and other stdlib packages.

**Comment by dopplershift** [comment]
It took me forever to find it, but here's the comment that made me think `micropip` was the proper way: https://github.com/pyodide/pyodide/discussions/2930#discussioncomment-3418535

Otherwise, you definitely need to document the hell out of what the proper way should be. 😉

**Comment by ryanking13** [comment]
`micropip.install("lzma")` works now in latest main branch (changed in #2946) but I think it was not vendored in 0.21.1.

**Comment by ryanking13** [comment]
> Otherwise, you definitely need to document the hell out of what the proper way should be. 😉

Right, sorry for the confusion. I forgot to update docs in #2946. It will be updated in #2998.

**Comment by ryanking13** [comment]
> We do want these to be installable with micropip, right? cc @ryanking13 I have not considered this issue during review.
> Or was the original intention to avoid shadowing identically named PyPI packages?

Yes, it should be installable with micropip. The error `InvalidRequirement: Parse error at "'_lzma'": Expected W:(0-9A-Za-z)` happens because `packaging.REQUIREMENT` doesn't allow the leading underscore in the package name, which I wasn't aware of.

**Comment by rth** [comment]
So I think the solution for 0.20.X is to rename back these stdlib packages to be without the leading underscore.

**Comment by ryanking13** [comment]
> So I think the solution for 0.21.X is to rename back these stdlib packages to be without the leading underscore.

Yes, it is already renamed on the main branch so we can release it with 0.21.2. Though we still need to fix the problem that micropip denies packages with a leading underscore. A package with a leading underscore is not allowed for PyPI packages but maybe we can have some other unvendored stdlibs with leading underscores in the future.

**Comment by rth** [comment]
> . A package with a leading underscore is not allowed for PyPI packages but maybe we can have some other unvendored stdlibs with leading underscores in the future.

It's not just PyPi, it's the Python package spec as enforced by `packaging`, so it would be better to follow that unless we absolutely need to have "private" packages.

**Comment by ryanking13** [comment]
> It's not just PyPi, it's the Python package spec as enforced by `packaging`, so it would be better to follow that unless we absolutely need to have "private" packages.

I think in the future, we might need those "private" packages for some C/C++ extensions of stdlib modules that have leading underscores.

For example, `_hashlib` is not included in the current version of Pyodide because it depends on `openssl`. Since we now have openssl,  we can think of unvendoring it. The problem is that `_hashlib` is used internally in `hashlib` but we don't want to unvendor `hashlib` together because `_hashlib` is not mandatory for `hashlib`. So we may need to unvendor `_hashlib` only. In that case, `micropip.install("_hashlib")` might need to be available in the future.

**Comment by rth** [comment]
This should be now fixed in 0.21.2 thanks to @ryanking13 
```
>>> import micropip
>>> await micropip.install('lzma')
```
